1-js/02-first-steps/04-variables/article.md

```diff
@@ -1,343 +1,348 @@
 
 ```js
 let userName;
 let test123;
 ```
 
-When the name contains multiple words, [camelCase](https://en.wikipedia.org/wiki/CamelCase) is commonly used. That is: words go one after another, each word except first starting with a capital letter: `myVeryLongName`.
+जब नाम में कई शब्द होते हैं, तो [camelCase](https://en.wikipedia.org/wiki/CamelCase) आमतौर पर इस्तेमाल किया जाता है। वह यह है: शब्द एक के बाद एक, प्रत्येक शब्द को छोड़कर पहले एक बड़े अक्षर से शुरू होते हैं: `myVeryLongName`।
 
-What's interesting -- the dollar sign `'$'` and the underscore `'_'` can also be used in names. They are regular symbols, just like letters, without any special meaning.
+क्या दिलचस्प है -- डॉलर चिह्न `'$'` और अंडरस्कोर `'_'` नामों में भी इस्तेमाल किया जा सकता है। वे बिना किसी विशेष अर्थ के, अक्षरों की तरह नियमित प्रतीक हैं।
 
-These names are valid:
+ये नाम मान्य हैं:
 
 ```js run untrusted
-let $ = 1; // declared a variable with the name "$"
-let _ = 2; // and now a variable with the name "_"
+let $ = 1; // "$" नाम के साथ एक variable घोषित
+let _ = 2; // और अब "_" नाम के साथ एक variable
 
 alert($ + _); // 3
 ```
 
-Examples of incorrect variable names:
+गलत variable नामों के उदाहरण:
 
 ```js no-beautify
-let 1a; // cannot start with a digit
+let 1a; // एक अंक से शुरू नहीं हो सकता
 
-let my-name; // hyphens '-' aren't allowed in the name
+let my-name; // नाम में '-' की अनुमति नहीं है
 ```
 
 ```smart header="Case matters"
-Variables named `apple` and `AppLE` are two different variables.
+`apple` और `AppLE` नाम के variables दो अलग-अलग variables हैं।
 ```
 
-````smart header="Non-Latin letters are allowed, but not recommended"
-It is possible to use any language, including cyrillic letters or even hieroglyphs, like this:
+````smart header="गैर-लैटिन अक्षरों की अनुमति है, लेकिन अनुशंसित नहीं है"
+किसी भी भाषा का उपयोग करना संभव है, जिसमें cyrillic अक्षरों या यहां तक ​​कि चित्रलिपि भी शामिल हैं:
 
 ```js
 let имя = '...';
 let 我 = '...';
 ```
 
-Technically, there is no error here. Such names are allowed, but there is an international convention to use English in variable names. Even if we're writing a small script, it may have a long life ahead. People from other countries may need to read it some time.
+तकनीकी रूप से, यहां कोई त्रुटि नहीं है। ऐसे नामों की अनुमति है, लेकिन variable नामों में अंग्रेजी का उपयोग करने के लिए एक अंतर्राष्ट्रीय सम्मेलन है। यहां तक ​​कि अगर हम एक छोटी सी स्क्रिप्ट लिख रहे हैं, उसका आगे का लंबा जीवन हो सकता है। अन्य देशों के लोगों को इसे कुछ समय पढ़ने की आवश्यकता हो सकती है।
 ````
 
 ````warn header="Reserved names"
-There is a [list of reserved words](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords), which cannot be used as variable names because they are used by the language itself.
+[आरक्षित शब्दों की एक सूची] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords) है, जिसका उपयोग उन variable नामों के रूप में नहीं किया जा सकता है, जिनका वे उपयोग करते हैं।
```

Comment:
We should put a better context and improve the sentence.

Reply:
Tried.